src/Geometry/Primitives/AsphericSurface.jl

```diff
@@ -0,0 +1,344 @@
+        @assert semidiameter > 0
+        @assert !isnan(semidiameter) && !isnan(radius) && !isnan(conic)
+        @assert one(T) - (1 / radius)^2 * (conic + one(T)) * semidiameter^2 > 0 "Invalid surface (conic/radius combination: $radius, $conic)"
+
+        acs = []
+        if aspherics===nothing
+            M = CONIC
+         else
+            asphericTerms = [i for (i, ) in aspherics]
+            minAsphericTerm = minimum(asphericTerms)
+            maxAsphericTerm = maximum(asphericTerms)
+            @assert minAsphericTerm > 0 "Aspheric Terms must be Order 1 or higher ($minAsphericTerm)"
+            acs = zeros(T, maxAsphericTerm )
+            for (i, k) in aspherics
+                acs[i] = k
+            end
+            odd = any([isodd(i) && a != zero(T) for (i, a) in enumerate(acs)])
+            even = any([iseven(i) && a != zero(T) for (i, a) in enumerate(acs)])
+            if odd && even
+                M = ODDEVEN
+            elseif even
+                M = EVEN
+                acs = [acs[2i] for i in 1:(maxAsphericTerm ÷ 2)] 
+            elseif odd
+                M = ODD
+                acs = [acs[2i-1] for i in 1:((maxAsphericTerm+1) ÷ 2)]
+            else #there are no nonzero aspherics terms in the list
+                M = CONIC
+                acs = []
+            end
+        end
+        Q = length(acs)
+        surf = new{T,3, Q, M}(semidiameter, 1/radius, conic, acs, normradius, Cylinder(semidiameter, interface = opaqueinterface(T)))
+        return surf
+    end
+
+end
+
+"""
+
+```julia
+EvenAsphericSurface(semidiameter, curvature::T, conic::T, aspherics::Vector{T}; normradius::T=semidiameter)
+```
+
+Surface incorporating an aspheric polynomial - radius, conic and aspherics are defined relative to absolute semi-diameter.
+
+`aspherics` should be an array of the even coefficients of the aspheric polynomial starting with A2
+
+
+"""
+function EvenAsphericSurface(semidiameter::T, curvature::T, conic::T, aspherics::Vector{T}; normradius::T=semidiameter) where T<:Real
+    Q=length(aspherics)
+    AsphericSurface(EVEN, semidiameter, curvature, conic, SVector{Q,T}(aspherics), normradius, Cylinder(semidiameter, interface = opaqueinterface(T)))
+end
+
+"""
+
+```julia
+OddAsphericSurface(semidiameter, curvature::T, conic::T, aspherics::Vector{T}; normradius::T=semidiameter)
+```
+
+Surface incorporating an aspheric polynomial - radius, conic and aspherics are defined relative to absolute semi-diameter.
+
+`aspherics`  should be an array of the odd coefficients of the aspheric polynomial starting with A1
+
+"""
+function OddAsphericSurface(semidiameter::T, curvature::T, conic::T, aspherics::Vector{T}; normradius::T=semidiameter) where T<:Real
+    Q=length(aspherics)
+    AsphericSurface(ODD, semidiameter, curvature, conic, SVector{Q,T}(aspherics), normradius, Cylinder(semidiameter, interface = opaqueinterface(T)))
+end
+
+"""
+
+```julia
```

Comment:
You don't need \`\`\`Julia...\`\`\` here, just indent the first line and it will be formatted correctly in the docs (same in other docstrings)

Reply:
You can check the docstrings of other function/classes to get an idea of the formatting